TRAMS-API/Repositories/AcademiesRepository.cs

```diff
@@ -8,25 +8,23 @@
 
 namespace API.Repositories
 {
-    public class AcademiesRepository
+    public class AcademiesRepository : IAcademiesRepository
     {
         private static readonly string _route = "accounts";
 
         private readonly AuthenticatedHttpClient _client;
-        private readonly IODataModelHelper<GetAcademiesD365Model> _oDataHelper;
+        private readonly IOdataUrlBuilder<GetAcademiesD365Model> _urlBuilder;
 
-        public AcademiesRepository(AuthenticatedHttpClient client, IODataModelHelper<GetAcademiesD365Model> oDataHelper)
+        public AcademiesRepository(AuthenticatedHttpClient client, 
+                                   IOdataUrlBuilder<GetAcademiesD365Model> urlBuilder)
         {
             _client = client;
-            _oDataHelper = oDataHelper;
+            _urlBuilder = urlBuilder;
         }
 
         public async Task<GetAcademiesD365Model> GetAcademyById(Guid id)
```

Comment:
The TrustRepository calls _client.AuthenticateAsync() but this one doesn't? Is it required?

Reply:
Yes, both should call the authentication method. Good spot!